torch/distributed/pipelining/schedules.py

```diff
@@ -1069,11 +1069,16 @@ def __init__(
         stage_index_to_group_rank: Optional[Dict[int, int]] = None,
         use_full_backward: bool = True,
     ):
-        if len(stages) <= 1:
-            raise ValueError(
-                f"Multi-stage schedule expects at least two stages but got {len(stages)}"
-            )
+
+        # TODO(whc) can we just delete this assert? It is convenient to do a unit test with a single stage, and
```

Comment:
> It is convenient to do a unit test with a single stage

Let's say this can this be implemented, we are not replacing all tests with single stage, right?

Reply:
No, I just added one new test. In general I found it's worth making tests single process and as simple as possible to test the desired behavior even if I could just use a multiproc multistage test. But there are still e2e multiproc tests for all the schedules.